### PR TITLE
Ifpack2: Use Kokkos::Experimental::partition_space instead of cudaStream

### PR DESCRIPTION
**Pull request by kliegeois**
@trilinos/ifpack2 

## Motivation

This PR updates the Block Tridiagonal Solver to use `Kokkos::Experimental::partition_space` instead of `cudaStream`.
This PR has been tested on V100 and MI250.
The performance on V100 are the same as the ones of the previous implementation.

### CODE REVIEW

File: packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp [hunk 1]
```diff
@@ -554,11 +554,12 @@ namespace Ifpack2 {
       local_ordinal_type_1d_view dm2cm; // permutation
 
 #if defined(KOKKOS_ENABLE_CUDA)
```
Comment (lucbv) [inline packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp:556]: Removing this `#if` block would be nice

File: packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp [hunk 2]
```diff
@@ -659,38 +660,24 @@ namespace Ifpack2 {
...
       void destroyExecutionSpaceInstances() {
-#if defined(KOKKOS_ENABLE_CUDA)
-        {
-          const local_ordinal_type num_streams = stream.size();
-          for (local_ordinal_type i=0;i<num_streams;++i)
-            KOKKOS_IMPL_CUDA_SAFE_CALL(cudaStreamDestroy(stream[i]));
-        }
-        stream.clear();
-        exec_instances.clear();
+#if defined(KOKKOS_ENABLE_CUDA) || defined(KOKKOS_ENABLE_HIP)
```
Comment (lucbv) [inline packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp:679]: Can probably get rid of this and just comment that Kokkos will clean thing up for us

**Review by lucbv — APPROVED** [review]
No issue with this in general, just a little clean up that we can tackle later on if needed

**Review by lucbv — APPROVED** [review]
Looks good to me, thanks for the clean up : )

**Comment by kliegeois** [comment]
Merging after double checking the performance on V100.

**Pull request merged**